its/src/test/java/com/sonar/it/scanner/msbuild/ScannerMSBuildTest.java

```diff
@@ -907,13 +910,57 @@ public void testAzureFunctions_WithWrongBaseDirectory_AnalysisSucceeds() throws
     if (VstsUtils.isRunningUnderVsts()) {
       // this might fail if Azure changes the drive
       assertThat(buildResult.getLogs()).contains("Using longest common projects path as a base directory: 'C:\\'");
-    }
-    else {
+    } else {
       String temporaryFolderRoot = temp.getRoot().getParent();
       assertThat(buildResult.getLogs()).contains("Using longest common projects path as a base directory: '" + temporaryFolderRoot + "'");
     }
   }
 
+  @Test
+  public void incrementalPrAnalysis_NoCache() throws IOException {
+    String projectKey = "incremental-pr-analysis-no-cache";
+    Path projectDir = TestUtils.projectDir(temp, "IncrementalPRAnalysis");
+    File unexpectedUnchangedFiles = new File(projectDir.resolve(".sonarqube\\conf\\UnchangedFiles.txt").toString());
+    BuildResult result = ORCHESTRATOR.executeBuild(TestUtils.newScanner(ORCHESTRATOR, projectDir)
+      .addArgument("begin")
+      .setProjectKey(projectKey)
+      .setProperty("sonar.projectBaseDir", projectDir.toAbsolutePath().toString())
+      .setDebugLogs(true) // To assert debug logs too
+      .setProperty("sonar.pullrequest.base", "base-branch"));
+
+    assertTrue(result.isSuccess());
+    assertThat(result.getLogs()).contains("Processing analysis cache");
+    assertThat(result.getLogs()).contains("Processing pull request with base branch 'base-branch'.");
+    assertThat(result.getLogs()).contains("Cache data is not available. Incremental PR analysis is disabled.");
+    assertThat(unexpectedUnchangedFiles.exists()).isFalse();
```

Comment:
SonarLint suggestion :)

```suggestion
    assertThat(unexpectedUnchangedFiles).doesNotExist();
```